omod/src/main/web/dashboardwidgets/latestobsforconceptlist/latestobsforconceptlist.controller.js

```diff
@@ -6,56 +6,38 @@ export default class LatestObsForConceptListController {
     }
 
     $onInit() {
-        this.maxAgeInDays = undefined;
-        this.obs = [];
         this.openmrsRest.setBaseAppPath("/coreapps");
         this.maxAgeInDays = this.widgetsCommons.maxAgeToDays(this.config.maxAge);
 
-        // Number of latest observations
-        let n_LatestObs = !angular.isUndefined(this.config.nLatestObs) && this.config.nLatestObs != null ? this.config.nLatestObs : 1;
-
         // Fetch last obs or obsGroup for the list of concepts
         this.openmrsRest.list('latestobs', {
             patient: this.config.patientUuid,
-            v: 'full',
+            v: 'custom:(' +
+                'concept:(uuid,display,datatype:(uuid),names:(name,locale,localePreferred,voided,conceptNameType)),' +
+                'value:(uuid,display,names:(name,locale,localePreferred,voided,conceptNameType)),' +
+                'groupMembers:(concept:(display,names:(name,locale,localePreferred,voided,conceptNameType))))',
             concept: this.config.concepts.split(',').map(c => c.trim()).join(','),
-            nLatestObs: n_LatestObs
+            nLatestObs: this.config.nLatestObs || 1
         }).then((resp) => {
             // Process the results from the list of concepts as not all of them may have data
-            for (let i = 0; i < resp.results.length; i++) {
-                let obs = resp.results[i];
+            this.obs = resp.results.filter(
                 // Don't add obs older than maxAge
-                if (angular.isUndefined(this.maxAgeInDays) || this.widgetsCommons.dateToDaysAgo(obs.obsDatetime) <= this.maxAgeInDays) {
-                    // Add last obs for concept to list
-                	
-                	if (angular.isDefined(obs.groupMembers) && obs.groupMembers != null) {
-                        // If obs is obs group
-                        let members = [];
-                        angular.forEach(obs.groupMembers, (member) => {
-                        	let prefix;
-                            let value;                            
-                        	
-                        	// Formatting the obs with concept prefix
-                        	if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "FSN") {
-                        		prefix = "(" + member.concept.display + ") ";
-                        	} else if (angular.isDefined(this.config.obsGroupLabels) && this.config.obsGroupLabels == "shortName") {
-                        		prefix = "(" + member.concept.name.display + ") ";
```

Comment:
I'm not sure how this was supposed to work. So I implemented it differently, by referring explicitly to the `names` list.